service/frontend/fx.go

```diff
@@ -163,29 +163,33 @@ func GrpcServerOptionsProvider(
 		logger.Fatal("creating gRPC server options failed", tag.Error(err))
 	}
 	interceptors := []grpc.UnaryServerInterceptor{
-		namespaceLogInterceptor.Intercept,
+		// Service Error Interceptor should be the most outer interceptor on error handling
 		rpc.ServiceErrorInterceptor,
+		namespaceValidatorInterceptor.SizeValidationIntercept,
+		namespaceLogInterceptor.Intercept, // TODO: Deprecate this with a outer custom interceptor
 		grpc.UnaryServerInterceptor(traceInterceptor),
 		metrics.NewServerMetricsContextInjectorInterceptor(),
-		retryableInterceptor.Intercept,
 		telemetryInterceptor.Intercept,
-		namespaceValidatorInterceptor.Intercept,
-		namespaceCountLimiterInterceptor.Intercept,
-		namespaceRateLimiterInterceptor.Intercept,
-		rateLimitInterceptor.Intercept,
 		authorization.NewAuthorizationInterceptor(
 			claimMapper,
 			authorizer,
 			metricsClient,
 			logger,
 			audienceGetter,
 		),
+		namespaceValidatorInterceptor.StateValidationIntercept,
+		namespaceCountLimiterInterceptor.Intercept,
+		namespaceRateLimiterInterceptor.Intercept,
+		rateLimitInterceptor.Intercept,
 		sdkVersionInterceptor.Intercept,
 		callerInfoInterceptor.Intercept,
 	}
 	if len(customInterceptors) > 0 {
+		// TODO: Deprecate WithChainedFrontendGrpcInterceptors and provide a inner custom interceptor
 		interceptors = append(interceptors, customInterceptors...)
 	}
+	// retry interceptor should be the most inner interceptor
+	interceptors = append(interceptors, retryableInterceptor.Intercept)
```

Comment:
should change [this](https://github.com/temporalio/temporal/blob/c6fb6b8331599b2b43b6da424224b26cb9d46904/service/fx.go#L105) as well

Reply:
Updated